File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/cesar-ch.js
// Operadores Aritméticos
let a = 5;
let b = 2;
console.log("Suma:", a + b);
console.log("Resta:", a - b);
console.log("Multiplicación:", a * b);
console.log("División:", a / b);
console.log("Módulo:", a % b);
console.log("Incremento:", ++a);
console.log("Decremento:", --b);

// Operadores de Comparación
let x = 10;
let y = "10";
console.log("Igualdad:", x == y);
console.log("Igualdad estricta:", x === y);
console.log("Desigualdad:", a !== b);
console.log("Mayor que:", a > b);
console.log("Menor que o igual:", a <= b);

// Operadores Lógicos
let p = true;
let q = false;
console.log("AND lógico:", p && q);
console.log("OR lógico:", p || q);
console.log("NOT lógico:", !p);

// Operadores de Asignación
let c = 3;
c += 2;
console.log("Asignación con adición:", c);

// Operadores de Identidad
console.log("Identidad:", x === 10);
console.log("No identidad:", y !== 10);

// Operadores de Pertenencia
let arreglo = [1, 2, 3];
console.log("Pertenece al arreglo:", 2 in arreglo);

// Operadores de Bits
let num1 = 5; // Representación binaria: 0101
let num2 = 3; // Representación binaria: 0011
console.log("AND a nivel de bits:", num1 & num2); // Resultado: 0001 (1 en binario)
console.log("OR a nivel de bits:", num1 | num2); // Resultado: 0111 (7 en binario)
console.log("Desplazamiento a la izquierda:", num1 << 1); // Resultado: 1010 (10 en binario)
console.log("Desplazamiento a la derecha:", num1 >> 1); // Resultado: 0010 (2 en binario)

// Estructuras de Control
// Condicionales
let edad = 18;
if (edad >= 18 && edad <= 125) {
    console.log("Eres mayor de edad");
} else if (edad < 18) {
    console.log("Eres menor de edad");
} else {
    console.log("Edad no valida");

}

// Excepciones
try {
    let resultado = 10 / 0;
    console.log("Resultado:", resultado);
} catch (error) {
    console.log("Error:", error.message);
} finally {
    console.log("Finalizado el bloque try-catch");
}

// Programa
for (let i = 10; i <= 55; i++) {
    if (i % 2 === 0 && i !== 16 && i % 3 !== 0) {
        console.log(i);
    }
}

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/c++/cesar-ch.cpp
/*
 * EJERCICIO:
 * - Crea ejemplos utilizando todos los tipos de operadores de tu lenguaje:
 *   Aritméticos, lógicos, de comparación, asignación, identidad, pertenencia, bits...
 *   (Ten en cuenta que cada lenguaje puede poseer unos diferentes)
 * - Utilizando las operaciones con operadores que tú quieras, crea ejemplos
 *   que representen todos los tipos de estructuras de control que existan
 *   en tu lenguaje:
 *   Condicionales, iterativas, excepciones...
 * - Debes hacer print por consola del resultado de todos los ejemplos.
 *
 * DIFICULTAD EXTRA (opcional):
 * Crea un programa que imprima por consola todos los números comprendidos
 * entre 10 y 55 (incluidos), pares, y que no son ni el 16 ni múltiplos de 3.
 *
 * Seguro que al revisar detenidamente las posibilidades has descubierto algo nuevo.
 */
#include <iostream>
using namespace std;

int main()
{
    // Operadores aritméticos
    int a = 10, b = 5;
    int suma = a + b;
    int resta = a - b;
    int multiplicacion = a * b;
    int division = a / b;
    int modulo = a % b;

    cout << "Operadores aritméticos:" << endl;
    cout << "Suma: " << suma << endl;
    cout << "Resta: " << resta << endl;
    cout << "Multiplicación: " << multiplicacion << endl;
    cout << "División: " << division << endl;
    cout << "Módulo: " << modulo << endl;

    // Operadores lógicos
    bool x = true, y = false;
    bool and_logico = x && y;
    bool or_logico = x || y;
    bool not_logico = !x;

    cout << "\nOperadores lógicos:" << endl;
    cout << "AND lógico: " << and_logico << endl;
    cout << "OR lógico: " << or_logico << endl;
    cout << "NOT lógico: " << not_logico << endl;

    // Operadores de comparación
    bool igualdad = (a == b);
    bool desigualdad = (a != b);
    bool mayor_que = (a > b);
    bool menor_que = (a < b);
    bool mayor_igual = (a >= b);
    bool menor_igual = (a <= b);

    cout << "\nOperadores de comparación:" << endl;
    cout << "Igualdad: " << igualdad << endl;
    cout << "Desigualdad: " << desigualdad << endl;
    cout << "Mayor que: " << mayor_que << endl;
    cout << "Menor que: " << menor_que << endl;
    cout << "Mayor o igual: " << mayor_igual << endl;
    cout << "Menor o igual: " << menor_igual << endl;

    // Operadores de asignación
    int c = 10;
    c += 5;
    cout << "\nOperador de asignación:" << endl;
    cout << "Valor de c después de c += 5: " << c << endl;
    c -= 5;
    cout << "Valor de d después de c -= 5: " << c << endl;
    c *= 5;
    cout << "Valor de c después de c *= 5: " << c << endl;
    c /= 5;
    cout << "Valor de c después de c /= 5: " << c << endl;
    c %= 5;
    cout << "Valor de c después de c %= 5: " << c << endl;

    // Operadores a nivel de bits
    cout << "\nOperadores a nivel de Bits: \n";
    cout << "a & b = " << (a & b) << endl; // AND a nivel de bits
    cout << "a | b = " << (a | b) << endl; // OR a nivel de bits
    cout << "a ^ b = " << (a ^ b) << endl; // XOR a nivel de bits

    // Estructuras de control (Condicionales)
    cout << "\nEstructura de Control - Condicional: \n";
    if (a > b)
    {
        cout << "a es mayor que b\n";
    }
    else
    {
        cout << "a no es mayor que b\n";
    }

    // Estructuras de control (Iterativas)
    cout << "\nEstructura de Control - Bucle For: \n";
    for (int i = 0; i < 5; i++)
    {
        cout << "i = " << i << endl;
    }

    // Manejo de excepciones
    cout << "\nEstructura de Control - Manejo de Excepciones: \n";
    try
    {
        if (b == 0)
            throw "Error: División por cero";
        cout << "a / b = " << a / b << endl;
    }
    catch (const char *msg)
    {
        cout << msg << endl;
    }

    // DIFICULTAD EXTRA
    cout << "\nDificultad Extra:\n";
    for (int i = 10; i <= 55; i++)
    {
        if (i % 2 == 0 && i != 16 && i % 3 != 0)
        {
            cout << i << " ";
        }
    }
    cout << endl;
}